metagpt/provider/together.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+from typing import Optional, Union, AsyncGenerator
+from together import Together
+
+from metagpt.provider.base_llm import BaseLLM
+from metagpt.configs.llm_config import LLMConfig, LLMType
+from metagpt.provider.llm_provider_registry import register_provider
+from metagpt.const import USE_CONFIG_TIMEOUT
+from metagpt.logs import logger
+from metagpt.utils.cost_manager import CostManager
+from metagpt.utils.token_counter import count_input_tokens, count_output_tokens, get_max_completion_tokens
+from metagpt.utils.exceptions import handle_exception
+
+@register_provider(LLMType.TOGETHER)
+class TogetherLLM(BaseLLM):
```

Comment:
Notice that together has openai api compatibility inferface `https://docs.together.ai/docs/openai-api-compatibility`. So it seems that no need to create a new provider here.